Task 4/Task4/Task4/UserCommunication.cs

```diff
@@ -0,0 +1,58 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+
+namespace Task4
+{
+    class UserCommunication
+    {
+        public void OpenMenu(CastomGit git, String SelectedStr)
+        {
+            Menu selected;
+
+            while(!Enum.TryParse(SelectedStr, out selected)) { }
```

Comment:
If you control your git from this menu, you shouldn't control menu from your git. Circular references can hit hard. Remember Dependency Inversion Principle from SOLID